tests/func-tests/monitoring_test.go

```diff
@@ -82,22 +104,39 @@ func getAlertByName(alerts promApiv1.AlertsResult, alertName string) *promApiv1.
 	return nil
 }
 
-func checkRunbookUrls(prometheusRule monitoringv1.PrometheusRule) {
-	for _, group := range prometheusRule.Spec.Groups {
-		for _, rule := range group.Rules {
-			if len(rule.Alert) > 0 {
-				ExpectWithOffset(1, rule.Annotations).ToNot(BeNil())
-				url, ok := rule.Annotations["runbook_url"]
-				ExpectWithOffset(1, ok).To(BeTrue())
-				checkAvailabilityOfUrl(url)
-			}
-		}
-	}
+func checkRunbookURL(rule monitoringv1.Rule) {
+	url, ok := rule.Annotations["runbook_url"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have runbook_url annotation", rule.Alert))
+	resp, err := runbookClient.Head(url)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("%s runbook is not available", rule.Alert))
+	ExpectWithOffset(1, resp.StatusCode).Should(Equal(200), fmt.Sprintf("%s runbook is not available", rule.Alert))
+}
+
+func checkSummary(rule monitoringv1.Rule) {
+	summary, ok := rule.Annotations["summary"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have summary annotation", rule.Alert))
```

Comment:
Please use built-in matcher; e.g.

```suggestion
	ExpectWithOffset(1, rule.Annotations).Should(HaveKey("summary"), fmt.Sprintf("%s does not have summary annotation", rule.Alert))
	ExpectWithOffset(1, rule.Annotations["summary"]).ShouldNot(BeEmpty(), fmt.Sprintf("%s has an empty summary", rule.Alert)
```

Reply:
This *may* will ***maybe*** also work:

```suggestion
	ExpectWithOffset(1, rule.Annotations).Should(HaveKeyWithValue("summary", Not(BeEmpty()), fmt.Sprintf("%s summery is missing or empty", rule.Alert))
```